api/base/utils.py

```diff
@@ -19,17 +20,24 @@ def absolute_reverse(view_name, query_kwargs=None, args=None, kwargs=None):
     return url
 
 
-def get_object_or_404(model_cls, query_or_pk):
+def get_object_or_error(model_cls, query_or_pk):
+
     if isinstance(query_or_pk, basestring):
         query = Q('_id', 'eq', query_or_pk)
     else:
         query = query_or_pk
+
     try:
-        return model_cls.find_one(query)
+        obj = model_cls.find_one(query)
+        if getattr(obj, 'is_deleted', False) is True:
+            raise Gone
+        if getattr(obj, 'is_active', None) is False:
```

Comment:
While the meaning of `is_deleted` is unambiguous, `is_active` may have different meanings, depending on the model. It's feasible to have a model with an `is_active` field where we wouldn't want to return a 410. We could err on the conservative side and have the view handle `is_active` instead of handling it here.

That said, since we only have one model that has `is_active` (User), I think this is fine to keep as is for now.